File: dbassistant/layers/LayerCargo.py
class LayerCargo:
    def __init__(self, cargo=None):
        self.cargo = cargo

    def put(self, value):
        self.cargo = value
        return self

    def take(self):
        if self.cargo is None:
            raise Exception("LayerCargo is empty")
        c = self.cargo
        self.cargo = None
        return c

File: dbassistant/layers/LayerCargo.ts
export default class LayerCargo {
    private cargo: any;

    constructor(cargo: any = null) {
        this.cargo = cargo;
    }

    put(value: any): LayerCargo {
        this.cargo = value;
        return this;
    }

    take(): any {
        if (this.cargo === null) {
            throw new Error("LayerCargo is empty");
        }
        const c = this.cargo;
        this.cargo = null;
        return c;
    }
}